aiida/restapi/translator/node.py

```diff
@@ -220,150 +348,253 @@ def get_results(self):
-            nodes.append({"id": 0,
-                          "nodeid":mainNode[0],
-                          "nodetype":mainNode[1],
-                          "group": "mainNode"
-                         })
-            addedNodes.append(mainNode[0])
-
-        # get all descendents
-        nodes, addedNodes, addedEdges = addNodes(nodeId, maxDepth, nodes,
-                               addedNodes, addedEdges, "ancestors")
-        nodes, addedNodes, addedEdges = addNodes(nodeId, maxDepth, nodes,
-                               addedNodes, addedEdges, "desc")
-
-        edges = addEdges(edges, addedNodes, addedEdges)
+            for input in qb.iterdict():
+
+                node = input['in']['*']
+                linktype = input['main--in']['label']
+                id = node.pk
+                nodetype = node.dbnode.type
+                display_type = nodetype.split('.')[-2]
+                description = get_additional_string(node)
+
+                nodes.append({
+                    "id": nodeCount,
+                    "nodeid": id,
+                    "nodetype": nodetype,
+                    "displaytype": display_type,
+                    "group": "inputs",
+                    "description": description,
+                    "linktype": linktype,
+                })
+                edges.append({
+                    "from": nodeCount,
+                    "to": 0,
+                    "arrows": "to",
+                    "color": {"inherit": 'from'},
+                    "linktype": linktype,
+                })
+                nodeCount += 1
+
+        # get all outputs
+        qb = QueryBuilder()
+        qb.append(Node, tag="main", project=['*'],
+                  filters=filter)
+        qb.append(Node, tag="out", project=['*'], edge_project=['label'],
+                  output_of='main')
+        if qb.count() > 0:
+            for output in qb.iterdict():
+
+                node = output['out']['*']
+                linktype = output['main--out']['label']
+                id = node.pk
+                nodetype = node.dbnode.type
+                display_type = nodetype.split('.')[-2]
+                description = get_additional_string(node)
+
+                nodes.append({
+                    "id": nodeCount,
+                    "nodeid": id,
+                    "nodetype": nodetype,
+                    "displaytype": display_type,
+                    "group": "outputs",
+                    "description": description,
+                    "linktype": linktype,
+                })
+                edges.append({
+                    "from": 0,
+                    "to": nodeCount,
+                    "arrows": "to",
+                    "color": {"inherit": 'from'},
+                    "linktype": linktype
+                })
+                nodeCount += 1
 
         return {"nodes": nodes, "edges": edges}
+
+
+# TODO Put these functions in some common place so to not repeat code.
```

Comment:
Could you open an issue to move this logic in a common place (and e.g. unify it with the `verdi graph generate` etc?
Actually, I think we should define a generic method that returns this in Node, and classes are allowed to override it. To think about formatting in various formats (ascii, latex, html, ...)
Probably we can have multiple functions, or a function with a parameter? I'm pretty sure there are people who already thought at this